python/hail/linalg/rowmatrix.py

```diff
@@ -0,0 +1,251 @@
+from hail.utils.java import Env, joption
+from hail.typecheck import *
+from hail.expr.expressions import expr_float64, matrix_table_source, check_entry_indexed
+
+row_matrix_type = lazy()
+
+
+class RowMatrix(object):
+    """Hail's row-distributed matrix of :py:data:`.tfloat64` elements.
+
+    .. include:: ../_templates/experimental.rst
+
+    Notes
+    -----
+    Row matrices must have fewer than :math:`2^31` columns.
+
+    Under the hood, row matrices are partitioned into group of rows.
+    """
+
+    def __init__(self, jrm):
+        self._jrm = jrm
+
+    @property
+    def n_rows(self):
+        """Number of rows.
+
+        Returns
+        -------
+        :obj:`int`
+        """
+        return self._jrm.nRows()
+
+    @property
+    def n_cols(self):
+        """Number of columns.
+
+        Returns
+        -------
+        :obj:`int`
+        """
+        return self._jrm.nCols()
+
+    @property
+    def shape(self):
+        """Shape of matrix.
+
+        Returns
+        -------
+        (:obj:`int`, :obj:`int`)
+           Number of rows and number of columns.
+        """
+        return self.n_rows, self.n_cols
+
+    @classmethod
+    @typecheck_method(entry_expr=expr_float64)
+    def from_entry_expr(cls, entry_expr):
+        """Creates a row matrix using a matrix table entry expression.
+
```

Comment:
Add an example.

Reply:
I added an example here and to the corresponding methods in BlockMatrix